src/WebJobs.Extensions.Http/AuthenticatedUser.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// A class containing information about the current authenticated user.
+    /// </summary>
+    public class AuthenticatedUser
```

Comment:
For reference, my prototype PRs were here:

- https://github.com/Azure/azure-webjobs-sdk-extensions/pull/262
- https://github.com/Azure/azure-functions-host/pull/1747

I was trying to avoid adding new types to represent users/claims, and instead use existing .NET Types, since this is all representable as core identity Types.

Reply:
So  ClaimsIdentityBindingProvider and ClaimsPrincipalBindingProvider were both cherry-picked in the commit from that first PR you posted. This new object contains a superset of those binding providers, and is non-.NET specific as Chris mentioned below.

However, adding this type does brings about it's own challenges, such as how to import the type itself, as well as maintenance/documentation. I am open to discussing ways around defining a new type, while still allowing cross language support and ideally providing access to more information than just the claims.

---
In reply to: [171758615](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/373#discussion_r171758615) [](ancestors = 171758615)